src/components/DuosDatePicker.tsx

```diff
@@ -10,22 +10,29 @@ import {
 import {AdapterDayjs} from '@mui/x-date-pickers/AdapterDayjs';
 import {Button} from '@mui/material';
 import DialogActions from '@mui/material/DialogActions';
-import {Dayjs} from 'dayjs';
+import dayjs, {Dayjs} from 'dayjs';
 import type {} from '@mui/x-date-pickers/themeAugmentation';
 
 
 interface DUOSDatePickerProps {
     inputFormat: string;
-    value: Dayjs;
+    defaultValue: Dayjs;
     onChange: any;
     onError: any;
     readOnly: boolean;
 }
 
 export const DuosDatePicker = (props: DUOSDatePickerProps) => {
-  const {inputFormat, value, onChange, onError, readOnly} = props;
+  const {inputFormat, defaultValue, onChange, onError, readOnly} = props;
   const duosColorBlue='#216FB4';
-
+  //Required to display the error on initialization with an invalid value when letting the date picker manage the value.
+  //onError must be excluded as a dependency of the hook because of change detection looping.
+  const checkInitialValue = useMemo(()=>{  if (defaultValue != null && defaultValue.toString() === 'Invalid Date') {
+    onError('Invalid Date', defaultValue.toString());
+  }
+  return true;},
+  // eslint-disable-next-line
+  [defaultValue]);
```

Comment:
I've always found the use of useMemo to be a bit confusing. Why is it helpful in this case?

Reply:
It's a small optimization to cache this outcome unless defaultValue changes.